src/k2/module/k23si/Module.cpp

```diff
@@ -268,6 +268,42 @@ dto::Key K23SIPartitionModule::_getContinuationToken(const IndexerIterator& it,
     };
 }
 
+// Makes the SKVRecord and applies the request's filter to it. If Status is not OK,
```

Comment:
why does bad status means that it should be returned?

Reply:
If the status is bad from doQueryFilter, it should be returned to the user in the query response. That is what I mean here, not sure if that answers your question.

Reply:
Oh i see. it is the status that is returned. I thought you're suggesting that the record should be returned if the status out of the filter is not OK.